java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/pub/ApiController.java

```diff
@@ -575,5 +574,13 @@ public String index(HttpServletRequest request, @RequestParam(required = false)
 	public String runLottery(@RequestParam(required = true) String latestHash) throws IOException {
 		return gson.toJson(lotteryPlayService.runLottery(latestHash));
     }
+	
+	@PostMapping(path = {"/promo-add-user"})
+	public ResponseEntity<String> addUser(@RequestParam String promoKey,
+	                                      @RequestParam String userEmail) {
+		synchronized (this) {
```

Comment:
this is absolutely the same as public synchronized ResponseEntity<String> addUser(@RequestParam String promoKey,
	                                      @RequestParam String userEmail)